src/senaite/jsonapi/api.py

```diff
@@ -1451,6 +1454,70 @@ def get_registry_records_by_keyword(keyword=None):
     return found_registers
 
 
+def get_settings_by_keyword(keyword=None):
+    """Get the settings associated to the specified keyword or, if
+     keyword is None, get all the settings.
+
+    :param keyword: settings to be retrieved
+    :return: dictionary with the settings plus a key to identify from which
+    keyword where retrieved.
+    """
+    key_to_ischema = {
+                      'mail':        [cp.mail.IMailSchema],
+                      'calendar':    [cp.calendar.ICalendarSchema],
+                      'ram':         [cp.ram.IRAMCacheSchema],
+                      'language':    [cp.language.ILanguageSelectionSchema],
+                      'editing':     [cp.editing.IEditingSchema],
+                      'usergroups':  [cp.usergroups.IUserGroupsSettingsSchema,
+                                      cp.usergroups.ISecuritySchema,],
+                      'search':      [cp.search.ISearchSchema],
+                      'filter':      [cp.filter.IFilterAttributesSchema,
+                                      cp.filter.IFilterEditorSchema,
+                                      cp.filter.IFilterSchema,
+                                      cp.filter.IFilterTagsSchema],
+                      'maintenance': [cp.maintenance.IMaintenanceSchema],
+                      'markup':      [cp.markup.IMarkupSchema,
+                                      cp.markup.ITextMarkupSchema,
+                                      cp.markup.IWikiMarkupSchema,],
+                      'navigation':  [cp.navigation.INavigationSchema],
+                      'security':    [cp.security.ISecuritySchema],
+                      'site':        [cp.site.ISiteSchema],
+                      'skins':       [cp.skins.ISkinsSchema],
+                      }
+
+    settings = []
+    if keyword is None:
+        for key, ischemas in key_to_ischema.items():
+            settings_for_key = get_settings_from_ischemas(ischemas)
+            settings.append({key: settings_for_key})
+    elif keyword in key_to_ischema.keys():
+        settings_from_key = get_settings_from_ischemas(key_to_ischema[keyword])
```

Comment:
here you can just use `get_settings_from_ischemas(keyword)`, because you know that the key exists in your mapping